plugins/inputs/jolokia2/jolokia.go

```diff
@@ -0,0 +1,220 @@
+package jolokia2
+
+import (
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Jolokia struct {
+	DefaultFieldPrefix    string
+	DefaultFieldSeparator string
+	DefaultTagPrefix      string
+
+	URLs            []string `toml:"urls"`
+	Username        string
+	Password        string
+	ResponseTimeout time.Duration `toml:"response_timeout"`
+
+	SSLCA              string `toml:"ssl_ca"`
+	SSLCert            string `toml:"ssl_cert"`
+	SSLKey             string `toml:"ssl_key"`
+	InsecureSkipVerify bool
+
+	Metrics []MetricConfig `toml:"metric"`
+}
+
+func (j *Jolokia) SampleConfig() string {
+	return `
+  # default_tag_prefix      = ""
+  # default_field_prefix    = ""
+  # default_field_separator = "."
+
+  ## Add agents to query
+  agents = ["http://localhost:8080/jolokia"]
+  # username = ""
+  # password = ""
+  # response_timeout = "5s"
+
+  ## Optional SSL config
+  # ssl_ca   = "/var/private/ca.pem"
+  # ssl_cert = "/var/private/client.pem"
+  # ssl_key  = "/var/private/client-key.pem"
+  # insecure_skip_verify = false
+
+  ## Add metrics to read
+  [[inputs.jolokia2.metric]]
+    name  = "java_runtime"
+    mbean = "java.lang:type=Runtime"
+    paths = ["Uptime"]
+`
+}
+
+func (j *Jolokia) Description() string {
+	return "Read JMX metrics from a Jolokia REST endpoint"
+}
+
+func (j *Jolokia) Gather(acc telegraf.Accumulator) error {
+	var metrics []Metric
+
+	for _, config := range j.Metrics {
+		metrics = append(metrics, NewMetric(config,
+			j.DefaultFieldPrefix, j.DefaultFieldSeparator, j.DefaultTagPrefix))
+	}
+
+	gatherer := NewGatherer(acc, metrics)
+	for _, url := range j.URLs {
+		agent, err := j.buildAgent(url)
+		if err != nil {
+			return err
+		}
+
+		err = gatherer.Gather(agent)
```

Comment:
This needs to be ran in a goroutine to avoid earlier urls blocking later ones.  It is not a problem in the JolokiaProxy since there is only one server, unless there are multiple targets.

You could get around this by saying there can only be one url per plugin, but the flip side of that is if you have many servers with the same configuration it would be a lot of config file to duplicate.